master/pkg/model/task_container_defaults.go

```diff
@@ -36,6 +36,7 @@ type TaskContainerDefaultsConfig struct {
 	Devices          DevicesConfig `json:"devices"`
 
 	BindMounts BindMountsConfig `json:"bind_mounts"`
+	WorkDir    *string          `json:"work_dir,omitempty"`
```

Comment:
I think the ideal world is that we will have `defaults.interactive_job.workdir` field. `InteractiveWorkDir` is confusing to me because it's unclear whether interactive describes the workdir or the jobs. I mean `InteractiveJobWorkDir` is better but it's too long and unnecessary. If people are going to set it, they should have already known what this field is for.

Reply:
> If people are going to set it, they should have already known what this field is for

That's not valid logic for naming something with an ambiguous name.  You could call it `abaiwoejfasoef` and say the same thing.

What about `command_work_dir`?

Reply:
Or even `notebook_work_dir`?

Reply:
This works for commands, notebooks, shells. I'd rather just use workdir. It's not end of the world to have a field that doesn't apply to experiments.

Reply:
No, it's a bad idea to introduce a name that we know is ambiguous.  I don't want to have to explain to OSS users who ask why their `workdir` isn't being honored that we decided on a knowingly-ambiguous name.

Reply:
Then just tell them to read the docs.